src/linearoperators/LBFGS.jl

```diff
@@ -1,132 +1,107 @@
 export LBFGS, update!
 
-# TODO make Ac_mul_B!
-# Edit: Ac_mul_B! is not really needed for this operator
-# Edit2: you never known! anyway for completeness would be cool to have it!
 """
-`LBFGS(T::Type, dim::Tuple, Memory::Int)`
+`LBFGS(x::Tuple, M::Integer)`
 
-`LBFGS{N}(T::NTuple{N,Type}, dim::NTuple{N,Tuple}, M::Int)`
-
-`LBFGS(x::AbstractArray, Memory::Int)`
+`LBFGS(x::AbstractArray, M::Integer)`
 
 Construct a Limited-Memory BFGS `LinearOperator` with memory `M`. The memory of `LBFGS` can be updated using the function `update!`, where the current iteration variable and gradient (`x`, `grad`) and the previous ones (`x_prev` and `grad_prev`) are needed:
 
 ```
 julia> L = LBFGS(Float64,(4,),5)
 LBFGS  ℝ^4 -> ℝ^4
 
-julia> update!(L,x,x_prev,grad,grad_prev); #update memory
+julia> update!(L,x,x_prev,grad,grad_prev); # update memory
 
-julia> d = L*x;                            #compute new direction
+julia> d = L*grad; # compute new direction
 
 ```
-
 """
 
-mutable struct LBFGS{M, N, R <: Real, T <: Union{R, Complex{R}}, A<:AbstractArray{T,N}} <: LinearOperator
-	currmem::Int
-	curridx::Int
-	s::A
-	y::A
-	s_m::NTuple{M, A}
-	y_m::NTuple{M, A}
-	ys_m::Array{R, 1}
+mutable struct LBFGS{R, T <: BlockArray, M} <: LinearOperator
+	currmem::Integer
+	curridx::Integer
+	s::T
+	y::T
+	s_M::Array{T, 1}
+	y_M::Array{T, 1}
+	ys_M::Array{R, 1}
 	alphas::Array{R, 1}
 	H::R
 end
 
-# Constructors
-#default
-function LBFGS(T::Type, dim::NTuple{N,Int}, M::Int) where {N}
```

Comment:
I would prefer to keep this constructor for consistency, this is the standard constructor for all operators:
`Operator(T::Type, dim::Tuple, args...)`